### PR TITLE
[model] feat: add GPT-OSS VeOmni backend

### PR DESCRIPTION
**Pull request by Luosuu**
## Summary
- Add GPT-OSS VeOmni model registration and patchgen-generated modeling.
- Wire FA4 sliding-window attention with learnable sinks.
- Add GPT-OSS MoE OpSlot variant with fail-fast non-eager dispatch and Quack fused MoE for native interleaved gate/up layout.
- Add GPT-OSS FLOPs/MFU accounting plus toy config and tests.
- Intentionally do not register GPT-OSS Triton fused MoE, because it does not support the native interleaved gate/up training path.

## Numerical Validation
- Fresh GPU re-run with explicit GPT-OSS MoE OpSlot rebinds validates FA4 + `fused_quack` across `fsdp`, `sp_fsdp`, `ep_fsdp`, and `ep_sp_fsdp`.
- E2E FA4 + `fused_quack` vs Transformers eager has stable logits diff across modes: `max=0.29296875`, `mean=0.002017322462052107`.
- The identical logits diff across modes is expected here: all modes compare the same VeOmni FA4 + `fused_quack` math against rank0 HF eager logits. In `fsdp`, distributed logits match the single-process VeOmni baseline exactly (`parallel_baseline_logits_max_diff=0`, `mean_diff=0`).
- Single-process kernel split shows the main e2e logits max-diff source appears when FA4 + learnable-sink attention is compared to Transformers eager, not when Quack MoE is toggled:
  - VeOmni eager attention + eager MoE: `max=0.01171875`, `mean=0.001538681099191308`.
  - FA4 attention + eager MoE: `max=0.3427734375`, `mean=0.0019805654883384705`.
  - Eager attention + `fused_quack`: `max=0.01171875`, `mean=0.0016568433493375778`.
  - FA4 attention + `fused_quack`: `max=0.29296875`, `mean=0.002017322462052107`.
- Additional sink probes show the large model-level FA4 diff is sink-dependent: zeroing all GPT-OSS attention sinks in both HF and VeOmni reduces FA4 + eager-MoE model diff to `max=0.0107421875`, `mean=0.0016573829343542457`.
- Direct FA4 cute varlen attention probes with explicit `learnable_sink`, GQA, causal local window, and multiple sink scales match eager reference closely (`max=0.0009765625`, `mean~=2.4e-5`) across `pack_gqa=None/True/False`.
- Layerwise exact replay with real model q/k/v/sinks shows FA4 direct/wrapper/model attention outputs match the correct causal/sliding eager reference closely (`pre-o max=0.00390625`, post-`o_proj` max `0.0009765625`). The direct FA4 path exactly matches the VeOmni wrapper/model path.
- The e2e max diff is amplified by GPT-OSS MoE top-k routing: router logits differ only slightly (`max=0.0078125`), but layer 1 has one top-k position mismatch, which is enough to produce the observed model-level max diff.
- The efficient fused MoE path remains very close to eager when attention is held fixed. Current evidence does not show wrong FA4 sink wiring (`s_aux -> learnable_sink`, shape, GQA head indexing, or mask/varlen dispatch); the observed max diff is from small math-equivalent FA4-vs-eager attention differences crossing a MoE routing boundary.

## Validation
- `ruff check veomni/models/transformers/gpt_oss/__init__.py veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py tests/models/test_gpt_oss_integration.py`
- `python tests/special_sanity/check_device_api_usage.py -d tests`
- `uv run --extra gpu --dev patchgen --check`
- `pytest tests/models/test_gpt_oss_integration.py -q -k "not cuda"`: `5 passed, 2 deselected`
- `pytest tests/models/test_model_registry.py -q -k gpt_oss`: `1 passed, 4 deselected`
- GPU: `pytest tests/models/test_gpt_oss_integration.py -q -k "fused_moe_matches_eager_cuda or quack_fused_moe_matches_eager_cuda" -s`: `2 passed, 5 deselected`
- GPU: GPT-OSS FA4 + `fused_quack` e2e parity re-run for `fsdp`, `sp_fsdp`, `ep_fsdp`, and `ep_sp_fsdp`.
- Earlier GPT-OSS parity/e2e coverage was validated, including logits parity, FSDP equivalence, and GPT-OSS FSDP/SP/EP e2e collection/gating.

## Review
- Latest PR comment/CI fixes address patched class registration, patchgen drift, device-api checks, no-eager-fallback behavior, and avoiding large `.contiguous()` copies on the Quack GEMM backward path.

### CODE REVIEW

File: veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+            grad_down_proj = gemm(fc1_activation.T, grad_fc2_output, cu_seqlens_k=cu_seqlens_m)
+
+        grad_gate_up = _gpt_oss_mlp_activation_backward(grad_fc1_activation, gate_up, ctx.alpha, ctx.limit)
+        grad_gate_up_proj_bias = _segment_sum(grad_gate_up, cu_seqlens_m)
+
+        scatter_output = moe_scatter(hidden_states, scatter_index)
+        grad_scatter_output = gemm(grad_gate_up, gate_up_proj.transpose(1, 2), cu_seqlens_m=cu_seqlens_m)
+        grad_gate_up_proj = None
+        if gate_up_proj.requires_grad:
+            grad_gate_up_proj = gemm(scatter_output.T, grad_gate_up, cu_seqlens_k=cu_seqlens_m)
```
[the remark this answers is not in the document]
Comment (Luosuu) [reply veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py:163]: this might lead to expensive large tensor copy which should be avoided. we can assert they are contiguous instead

File: veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+            grad_down_proj = gemm(fc1_activation.T, grad_output, cu_seqlens_k=cu_seqlens_m)
+
+        grad_gate_up = _gpt_oss_mlp_activation_backward(grad_fc1_activation, gate_up, ctx.alpha, ctx.limit)
+        grad_gate_up_proj_bias = _segment_sum(grad_gate_up, cu_seqlens_m)
+
+        grad_permute_tokens = gemm(grad_gate_up, gate_up_proj.transpose(1, 2), cu_seqlens_m=cu_seqlens_m)
+        grad_gate_up_proj = None
+        if gate_up_proj.requires_grad:
+            grad_gate_up_proj = gemm(permute_tokens.T, grad_gate_up, cu_seqlens_k=cu_seqlens_m)
+
```
[the remark this answers is not in the document]
Comment (Luosuu) [reply veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py:244]: same here. we can assert they are contiguous instead

File: veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+    gate = gate_raw.clamp(max=limit)
+    up = up_raw.clamp(min=-limit, max=limit)
+
+    sigmoid_gate = torch.sigmoid(gate * alpha)
+    grad_gate = grad_activation * (up + 1) * (sigmoid_gate + alpha * gate * sigmoid_gate * (1 - sigmoid_gate))
+    grad_gate = grad_gate * (gate_raw <= limit)
+
+    grad_up = grad_activation * gate * sigmoid_gate
+    grad_up = grad_up * ((up_raw >= -limit) & (up_raw <= limit))
+
```
[the remark this answers is not in the document]
Comment (Luosuu) [reply veomni/ops/kernels/moe/quack_gemm_interleave_gate_up.py:58]: we do not need to cover NPUs or older PyTorch versions. but making dtype explicit is a good point

**Review by FoolPlayer — APPROVED** [review]
